# Call Data

In [19]:
#clean up csv files
import pandas as pd

In [20]:
#variables for files
data = 'Data/clean_df.csv'
genre_data = 'Data/clean_genre.csv'

In [21]:
#import csvs
genre_data = pd.read_csv(genre_data)
data = pd.read_csv(data)

# Find Most Popular Genre

In [22]:
#Print genres dataframe
genre_data

Genres  Danceability (0-1)  Tempo (BPM)  \
0     21st century classical            0.284100    91.351000   
1                      432hz            0.312000   118.900933   
2                      8-bit            0.673000   192.816000   
3                 a cappella            0.516172   105.506031   
4                   abstract            0.473600   142.495400   
...                      ...                 ...          ...   
3226        zim urban groove            0.553000   134.995000   
3227                    zolo            0.533837   126.765194   
3228                    zouk            0.641889   105.848889   
3229            zurich indie            0.705667    91.278000   
3230                  zydeco            0.625963   130.137242   

      Popularity (0-100) Key   Mode  
0               6.600000   E  Major  
1              41.200000   B  Major  
2               0.000000   F  Major  
3              39.086248   G  Major  
4              43.800000   F  Major  
...                  ...  ..    ...  
3226            9.000000   G  Major  
3227           31.108254   A  Major  
3228           32.555556   G  Major  
3229            0.000000   G  Minor  
3230           27.703810   A  Major  

[3231 rows x 6 columns]

In [52]:
#Make genre column into string so all data is the same type
genre_data['Genres'] = genre_data['Genres'].apply(str)

#Sort Popularity in descending order to confirm most popular genre.
sorted_genre = genre_data.sort_values("Popularity (0-100)", ascending=False)
sorted_genre

Genres  Danceability (0-1)  Tempo (BPM)  Popularity (0-100)  \
564   chinese electropop               0.660      142.018                79.0   
1767  korean mask singer               0.658      128.909                78.0   
995        dutch rap pop               0.767      169.990                77.0   
3212                yaoi               0.603      135.028                77.0   
944              dong-yo               0.829      115.056                76.0   
...                  ...                 ...          ...                 ...   
2089           neo-kraut               0.705      123.993                 0.0   
573             chip hop               0.780       77.060                 0.0   
2283       pagode baiano               0.642      211.968                 0.0   
256               batida               0.861      119.984                 0.0   
823       dark psytrance               0.725      147.006                 0.0   

     Key   Mode  
564   Db  Minor  
1767   A  Minor  
995    F  Major  
3212  Ab  Minor  
944   Ab  Major  
...   ..    ...  
2089   A  Major  
573   Eb  Major  
2283   E  Major  
256    B  Major  
823    B  Minor  

[3231 rows x 6 columns]

With the cleaned and sorted data, we find that the most popular genre is Chinese Electropop.

In [59]:
#Sort Popularity in descending order to confirm most popular artist.
sorted_data = data.sort_values("Popularity (0-100)", ascending=False)
sorted_data = sorted_data.dropna()
sorted_data

Artists  Danceability (0-1)  \
20062                                      Olivia Rodrigo               0.585   
19862                               24kGoldn', 'iann dior               0.700   
19866                                       Ariana Grande               0.737   
19886                            Bad Bunny', 'Jhay Cortez               0.731   
19976                                             KAROL G               0.863   
...                                                   ...                 ...   
63064                                      Tarun Banerjee               0.530   
63063                                    Georges Derveaux               0.209   
63062                                    Эрнест Хемингуэй               0.800   
63061   Hector Berlioz', 'Concertgebouworkest', 'Eduar...               0.276   
174388                                          Roger Fly               0.677   

        Duration (seconds) Explicit Key   Mode  \
20062              242.014      Yes  Bb  Major   
19862              140.526      Yes   G  Minor   
19866              172.325      Yes   C  Major   
19886              205.090      Yes   E  Minor   
19976              178.947      Yes  Db  Minor   
...                    ...      ...  ..    ...   
63064              199.227       No   B  Minor   
63063              162.623       No   D  Major   
63062               95.700      Yes  Gb  Major   
63061              370.787       No   A  Major   
174388             197.710       No   G  Major   

                                               Song Title  Popularity (0-100)  \
20062                                     drivers license                 100   
19862                              Mood (feat. iann dior)                  96   
19866                                           positions                  96   
19886                                              DÁKITI                  95   
19976                                             BICHOTA                  95   
...                                                   ...                 ...   
63064                                        Phul Nebe Go                   0   
63063                                            Habanera                   0   
63062                          Часть 44.3 - Старик и море                   0   
63061   Symphonie fantastique, Op. 14, H 48: 2. Un bal...                   0   
174388                                     Improvisations                   0   

        Tempo (BPM)  Release Year  
20062       143.874          2021  
19862        90.989          2020  
19866       144.015          2020  
19886       109.928          2020  
19976       163.908          2020  
...             ...           ...  
63064        79.712          1952  
63063       139.475          1952  
63062        42.490          1952  
63061        89.437          1952  
174388      112.208          2020  

[174389 rows x 10 columns]

# Find Most Popular Artist

Currently Olivia Rodrigo is most popular with her song Drivers License. 

In [63]:
#To find out more about Olivia Rodrigo, we use loc to gather data regarding only her.
olivia_data =  sorted_data.loc[sorted_data["Artists"] == "Olivia Rodrigo", [
    "Artists", "Song Title", "Popularity (0-100)"]]
olivia_data

Artists                                         Song Title  \
20062   Olivia Rodrigo                                    drivers license   
19732   Olivia Rodrigo  All I Want - From "High School Musical: The Mu...   
110088  Olivia Rodrigo                                         All I Want   

        Popularity (0-100)  
20062                  100  
19732                   82  
110088                  75

All three of her songs available on Spotify are incredibly popular. 

In [64]:
#We found most popular artist based on the popularity of the song, but let us see which artist has the most songs listed.
sorted_data["Artists"].value_counts()

Tadeusz Dolega Mostowicz                                                             1281
Эрнест Хемингуэй                                                                     1175
Эрих Мария Ремарк                                                                    1062
Francisco Canaro                                                                      951
Ignacio Corsini                                                                       624
                                                                                     ... 
Stephen Sondheim', 'Len Cariou', 'Victor Garber', 'Merle Louise', 'Paul Gemignani       1
Bo Donaldson & The Heywoods                                                             1
Sunshine Anderson                                                                       1
UNKLE                                                                                   1
The Urge                                                                                1
Name: Arti

In [65]:
#To find out more about Francisco Canaro, we use loc to gather data regarding only him.
canaro =  sorted_data.loc[sorted_data["Artists"] == "Francisco Canaro", [
    "Artists", "Song Title", "Popularity (0-100)"]]
canaro

Artists                                         Song Title  \
4705   Francisco Canaro                   Soñar y Nada Mas - Remasterizado   
24208  Francisco Canaro                 Mas Solo Que Nunca - Remasterizado   
40111  Francisco Canaro              Mimosa - Instrumental (Remasterizado)   
24202  Francisco Canaro              Café de los Angelitos - Remasterizado   
21144  Francisco Canaro  La Rancherita de Santa Lucia - Instrumental (R...   
...                 ...                                                ...   
60702  Francisco Canaro    Tango Milonguero - Instrumental (Remasterizado)   
60681  Francisco Canaro  Bandoneón de mis amores - Instrumental (Remast...   
61187  Francisco Canaro                        Anoche Soñé - Remasterizado   
61185  Francisco Canaro                  Decile Que Vuelva - Remasterizado   
61168  Francisco Canaro             La Vida en 1000 Gramos - Remasterizado   

       Popularity (0-100)  
4705                   24  
24208                   4  
40111                   2  
24202                   2  
21144                   2  
...                   ...  
60702                   0  
60681                   0  
61187                   0  
61185                   0  
61168                   0  

[951 rows x 3 columns]

Tadeusz Dolega Mostowicz has the most titles on Spotify with a stunning 1,281 total. These titles, however, are snippets of audiobooks and not songs. The first music artist with the greatest amount of pieces is Francisco Canaro, a violinist from Uruguay. After generating a dataframe reflectig his popularity, the quantity of his titles does not display quality, his popularity far lower than Olivia Rodrigo's.

# Find the Artists Currently Most Popular per Decade

In [75]:
#Find range for years.
data.describe()

#create new call-name for dataframe
decade_data = data

#Minimum year is 1920, with max being 2021. Now we set our bins.
bins = [1920, 1929, 1939, 1949, 1959, 1969, 1979, 1989, 1999, 2009, 2019]

decade = ["1920s", "1930", "1940s", "1950s", "1960s", "1970s", "1980s", "1990s", "2000s", "2010s"]

#Using bins we create a column just for them.
decade_data["Decades"] = pd.cut(data["Release Year"], bins, labels=decade, include_lowest=True)

#Drop rows for 2020 and 2021, as these are for an incomplete decade.
decade_data.dropna()

Artists  Danceability (0-1)  Duration (seconds)  \
0                 Mamie Smith               0.598             168.333   
1       Screamin' Jay Hawkins               0.852             150.200   
2                 Mamie Smith               0.647             163.827   
3             Oscar Velazquez               0.730             422.087   
4                        Mixe               0.704             165.224   
...                       ...                 ...                 ...   
174116         Workout Music                0.734             277.493   
174118         Workout Music                0.533             287.987   
174120         Workout Music                0.764             268.120   
174124                October               0.520             226.603   
174128            DJ X-Change               0.620              30.250   

       Explicit Key   Mode                                         Song Title  \
0            No   F  Minor                           Keep A Song In Your Soul   
1            No   F  Minor                               I Put A Spell On You   
2            No   C  Major                                       Golfing Papa   
3            No   D  Major  True House Music - Xavier Santos & Carlos Gomi...   
4           Yes  Bb  Minor                                          Xuniverxe   
...         ...  ..    ...                                                ...   
174116       No   C  Minor                  RITMO (Bay Boys For Life) [Remix]   
174118       No   F  Major                           Beautiful People (Remix)   
174120       No   D  Major                                       Tusa (Remix)   
174124       No  Db  Major                      A Good Year - 2020 Remastered   
174128       No   C  Major              New Years Eve Ultimate Countdown 2020   

        Popularity (0-100)  Tempo (BPM)  Release Year Decades  
0                       12      149.976          1920   1920s  
1                        7       86.889          1920   1920s  
2                        4       97.600          1920   1920s  
3                       17      127.997          1920   1920s  
4                        2      122.076          1920   1920s  
...                    ...          ...           ...     ...  
174116                  14      128.015          2019   2010s  
174118                  14      129.943          2019   2010s  
174120                  16      127.994          2019   2010s  
174124                   5      123.976          2013   2010s  
174128                   8       93.056          2019   2010s  

[168255 rows x 11 columns]